File: app/src/main/java/dev/blackcat/minauta/net/FakeConnection.java
package dev.blackcat.minauta.net;

import java.util.Calendar;

import dev.blackcat.minauta.data.Account;
import dev.blackcat.minauta.data.Session;

public class FakeConnection implements Connection
{
    @Override
    public LoginResult login(Account account)
    {
        Session session = new Session();
        session.setLogoutParams("?x=x");
        session.setTimeParams("?x=x");
        session.setStartTime(Calendar.getInstance().getTimeInMillis());

        LoginResult result = new LoginResult();
        result.session = session;
        result.state = State.OK;

        return result;
    }

    @Override
    public LogoutResult logout(Account account)
    {
        LogoutResult result = new LogoutResult();
        result.state = State.OK;

        return result;
    }

    @Override
    public AvailableTimeResult getAvailableTime(Account account)
    {
        AvailableTimeResult result = new AvailableTimeResult();
        result.availableTime = "02:30:15";
        result.state = State.OK;

        return result;
    }
}

File: app/src/main/java/dev/blackcat/minauta/net/FakeConnection.kt
package dev.blackcat.minauta.net

import java.util.Calendar

import dev.blackcat.jnauta.net.AuthenticationResponseParser
import dev.blackcat.minauta.data.Account
import dev.blackcat.minauta.data.Session

open class FakeConnection : Connection {

    override suspend fun login(account: Account): Connection.LoginResult {
        val session = Session()
        session.loginParams = "x=x&y=y"
        session.startTime = Calendar.getInstance().timeInMillis

        val result = Connection.LoginResult()
        result.session = session
        result.state = Connection.State.OK
        return result
    }

    override suspend fun logout(account: Account, session: Session): Connection.LogoutResult {
        val result = Connection.LogoutResult()
        result.session = session
        result.state = Connection.State.OK
        return result
    }

    override suspend fun getAvailableTime(account: Account, session: Session): Connection.AvailableTimeResult {
        val result = Connection.AvailableTimeResult()
        result.availableTime = "00:00:20:time@fake.connection"
        result.state = Connection.State.OK
        return result
    }
}

class FakeNoLoginConnection: FakeConnection() {

    override suspend fun login(account: Account): Connection.LoginResult {
        val result = super.login(account)
        result.state = Connection.State.ERROR
        return result
    }

}

class FakeNoLogoutConnection: FakeConnection() {

    override suspend fun logout(account: Account, session: Session): Connection.LogoutResult {
        val result = super.logout(account, session)
        result.state = Connection.State.ERROR
        return result
    }

}
